### ISSUE TITLE
Use Next-auth to handle Oauth without authentication (Reopen #6577)

### ISSUE THREAD

**Issue by Khaaz** [nextauthjs/next-auth#6616]
### Question 💬

Hello,
I am sorry to re-open issue #6577 
I definitely don't understand your answer.

I am adding more context and reforumating my question here:

## What I want

I want a way to start the Oauth process (get the oauth token, and the Oauth data) for a Provider WITHOUT authentication.
Basically fetch a `Profile` for a specific Provider in itself. But I don't want to do that in the context of authentication, that create a session and so on.

In fact I want the authentication (so the normal next-auth use_case) in order for my user to login to my app.  
But I would also like to be able to make Oauth request independantly (for example when clicking on a button), fetching the oauth token (and then do subsequent request to the API I am talking to) without starting the authentication process and so on.

## Current understanding of solutions

Currently, the way I understand how next-auth work, I don't have a way to decouple the Oauth request from the authentication process.
So I would have to implement Oauth request myself, or use other libraries. I don't understand how to use next-auth to do that (which is a shame considering all OAuth process is already implemented in the lib).

Is it possible? What am I missing?

## Practical use case 

In practice:

- A User login to my app with discord (so he is logged-in with his token and profile linked with his discord profile)
- He can click on a button to start Oauth connection with another account (for instance twitter).
- This should not login-in / log-out the user in this case. He is still logged-in with Discord. We just retrieve information and a token from user connection, that we can for instance store in a variable, or in the session and that we then can use as we want.
- The use can now resume to normal activity, including doing the Oauth process with yet another provider for example

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
My answer is the same as https://github.com/nextauthjs/next-auth/issues/6577#issuecomment-1411322086. Adding an `adapter: YourAdapter()` (see the docs for built-in adapters or build your own.) and multiple providers, the flow you are describing will "just work". Accounts are created for each provider, that are connected to the same user in the database. You can also check out how it looks at https://next-auth.js.org/adapters/models.

It's unclear what you mean by:

> I want a way to start the Oauth process (get the oauth token, and the Oauth data) for a Provider WITHOUT authentication.
